staging/src/k8s.io/apiserver/pkg/server/httplog/httplog.go

```diff
@@ -125,10 +126,21 @@ func withLogging(handler http.Handler, stackTracePred StacktracePred, shouldLogR
 		rl := newLoggedWithStartTime(req, w, startTime)
 		rl.StacktraceWhen(stackTracePred)
 		req = req.WithContext(context.WithValue(ctx, respLoggerContextKey, rl))
-		defer rl.Log()
+		logInRoutine := false
+		defer func() {
+			if !logInRoutine {
+				rl.Log()
+			}
+		}()
 
 		w = responsewriter.WrapForHTTP1Or2(rl)
 		handler.ServeHTTP(w, req)
+
+		// log after the task is executed in the separate routine
+		// there is a task created in handler.ServeHTTP()
+		if routine.AppendTask(ctx, &routine.Task{Func: rl.Log}) {
```

Comment:
Hmm - when calling AppendTask, the main function is not yet set, right?
So won't it be overwritten later anyway?

I would like to see test output to see we actually fixed the problem.

Reply:
the main function is already set in the handler.ServeHTTP() above. I did test it and saw logs:

```
I0629 09:31:28.480093  980427 get.go:261] "Starting watch" path="/api/v1/namespaces/default/configmaps" resourceVersion="2031772" labels="" fields="" timeout="32m26.197309045s"
I0629 09:31:32.709322  980427 routine.go:45] "HTTP" verb="WATCH" URI="/api/v1/namespaces/default/configmaps?resourceVersion=2031772&watch=true" latency="4.229978453s" userAgent="kubectl/v1.27.15 (linux/amd64) kubernetes/a62bdc4" audit-ID="00868cb3-d8d2-4f07-b6d3-808ca095fb32" srcIP="127.0.0.1:56660" apf_pl="global-default" apf_fs="global-default1" apf_iseats=1 apf_fseats=0 apf_additionalLatency="0s" apf_init_latency="578.4µs" apf_execution_time="579.83µs" resp=200
```

Reply:
OK right - it's not called, but it's set.